romancal/datamodels/tests/test_models.py

```diff
@@ -1,16 +1,192 @@
-import asdf
+import pytest
 
-from romancal.datamodels import RomanDataModel
+import asdf
+import numpy as np
+from astropy.time import Time
 
+from romancal import datamodels
+from stdatamodels.validate import ValidationWarning
 
-def test_model_schemas():
-    def iter_subclasses(model_class):
+# Helper class to iterate over model subclasses
+def iter_subclasses(model_class, include_base_model=True):
+    if include_base_model:
         yield model_class
-        for sub_class in model_class.__subclasses__():
-            yield from iter_subclasses(sub_class)
+    for sub_class in model_class.__subclasses__():
+        yield from iter_subclasses(sub_class)
 
-    for model_class in iter_subclasses(RomanDataModel):
+def test_model_schemas(tmp_path):
+    for model_class in iter_subclasses(datamodels.RomanDataModel):
         try:
             asdf.schema.load_schema(model_class.schema_url)
         except Exception:
-            assert False, f"Unable to load schema URI '{model_class.schema_url}' for model class {model_class.__name__}"
+            assert False, f"Unable to load schema URI '{model_class.schema_url}' for model " \
+                          f"class {model_class.__name__}"
+
+
+# Helper function to test required keywords
+# meta = dictionary of required key/value pairs to ensure each are required
+# model_class = the class of the model to be tested
+def confirm_required_keywords(tmp_path, meta, model_class):
+    file_path = tmp_path / "test.asdf"
```

Comment:
What is `tmp_path` here and what is this line supposed to do?

Reply:
This needs to be removed, thanks for the catch. (It was for the code that Ed's suggestion replaced.)